### ISSUE TITLE
Import failed, path does not exist

### ISSUE THREAD

**Issue by jpborjas** [Radarr/Radarr#446]
I'm new to Radarr, current user of Sonarr and CouchPotato. Upon installing Radarr, I picked one available movie to be downloaded, just to test things out. Radarr downloaded in via SABnzbd, but now I'm getting this error:

17-1-24 19:00:22.5|Error|DownloadedMovieImportService|Import failed, path does not exist or is not accessible by Radarr: /Users/pborjas/Movies/Movies/Snowden.2016.1080p.BluRay.x264-RE.ENCODED/

The file is there inside a folder with the same name. I had turned on Set Permissions to 0755, and now I turned it off. I deleted everything and tried again, same error. The file and folders live on the SABNZBD server, and Sonarr and Radarr (and CouchPotato) on the server that houses all the media, using Plex. 

![screen shot 2017-01-25 at 8 25 52 am](https://cloud.githubusercontent.com/assets/14253583/22292156/f330cbcc-e2d7-11e6-9b39-4355a8029adb.png)

BTW, and I can open another case for this, if I drop a movie in the monitored folder, the one where the downloaded movies are processed from, Radarr is not even attempting to import them. 

Any ideas?

**Comment by hotio** [comment]
Radarr is looking in the folder:
/Users/pborjas/Movies/Movies/Snowden.2016.1080p.BluRay.x264-RE.ENCODED/

You have your file located in:
/Movies/Movies/Snowden.2016.720p.BluRay.DD5.1.x264-TayTo/

Seem pretty normal to me, that it can't find them. Would actually be suprised if it did find them.

**Comment by jpborjas** [comment]
Ah, funny!. I'm not THAT dumb. my screenshot shows the folder from the wrong server's point of view. However, point taken. I'm looking at it from the terminal (OS X).
![screen shot 2017-01-25 at 10 47 24 am](https://cloud.githubusercontent.com/assets/14253583/22297582/bb3ca43e-e2eb-11e6-9c9e-b5454030fc83.png)

I'm using the GUI to specify the folder, and this is what the GUI looks like for my Drone Folder. 

![screen shot 2017-01-25 at 10 47 24 am](https://cloud.githubusercontent.com/assets/14253583/22297672/05b45368-e2ec-11e6-8950-0103af601a36.png)

It looks like Radarr is getting the "/Users/pborjas....." path from SABNZBD, and that path is valid for that process on that particular server (different machine, IP, etc), but it is not a valid path for Sonarr, as the drive to look for finished downloaded movies is a mounted folder, as I specified in Drone. Is there a place that I can tell Sonarr to use the Drone folder and not SAB's? Is that what Remote Path Mappings is for?

Thanks for your reply.

**Comment by hotio** [comment]
Settings > Download Client > Remote Path Mappings

**Comment by jpborjas** [comment]
I disabled the Drone Factory and added the Remote Path. is this something that get's monitored, or something that get's triggered only upon download? Which brings me to my other question: Is there a folder that get's monitored, that if I downloaded a movie outside of Radarr, Raddar will pick it up, rename it and move it over the Movies Folder? Is that what the Drone Folder is for? I need that, and mine is on a remote server (mapped).

**Comment by galli-leo** [comment]
@jpborjas That is exactly what the drone factory folder is for. Additionally, if the previous import failed, it should try again every 30s.

**Comment by hotio** [comment]
From my understanding it's like this:

Sabnzbd downloads a file in:
/remote/movie
Radarr receives this information from sabnzbd and will look at:
/remote/movie
If you setup remote path mappings like "sabhost.com   /remote/movie    /local/movie":
It will look in:
/local/movie

It will keep looking in that folder and flood the log with errors until it finds that folder.

If Radarr has no idea that sabnzbd downloaded anything, you will have to use the Drone Factory.

**Comment by jpborjas** [comment]
The Remote Path Mapping seems to be working. Now, I just read a message that tells me pretty much that the Drone Factory Folder CAN'T be the same as the Remote Path, or the import will be skipped, and it will rely on the Drone for the download. Let me play with that for a minute, as there are a number of movies that I download by directly adding the nzb file to the watch folder. 

Thanks a lot for all the replies and all the hard work. The interface looks great! A little bit of a learning curve, over here. I'll update once I've figured it out.

**Comment by hotio** [comment]
If they are in sabnzbd changing the label to radarr might let radarr recognize them.

**Comment by jpborjas** [comment]
Label?, are you talking about Category? Where is that in Sab?

**Comment by jpborjas** [comment]
The Remote Path Mappings worked. I was able to Add a Movie and Radarr sent the request over to SAB, SAB downloaded it, and Radarr moved it over to my Plex Movie folder, with one hiccup: it didn't delete the folder SAB created. Also (two hiccups), it didn't copy any subtitle files. Can it do that, are there settings for these things?

**Comment by hotio** [comment]
No, it will ignore subtitles and for torrents it will also only copy the files, not sure what the correct behaviour should be for nzb. I just regularly let a script delete all files older than 2 days.

**Comment by jpborjas** [comment]
Feature requests?

I just downloaded another movie directly on SAB via the Watch Folder, configured the Drone Folder to monitor the folder (a remote folder), but nothing is hapening, no errors, nothing in the logs, nothing. 

Thoughts?

**Comment by hotio** [comment]
Have no experience with the Drone

**Comment by galli-leo** [comment]
@jpborjas Did you add the movie to Radarr before adding it to SAB? Because Radarr will only pick up movies in the drone folder it knows about.

**Comment by jpborjas** [comment]
Ah, I did not, but if I do that, here's what happens: I would have to tell Radarr to Not Monitor the movie, otherwise, it'll start to download it, right? I did that, and now is it telling me "Intermediate Download path inside drone factory, Skipping." Does that mean that the Drone folder can't be the same as the SAB folder for finished downloads? That would be a problem. I wouldn't want to tell Sab to put the movie somewhere else, or do it manually.

**Comment by jpborjas** [comment]
I'm creating a couple of enhancement requests. Thank you guys, great job so far!

**Comment by crockadile35** [comment]
Hello, i'm having some trouble here with the configuration.
I have sab  and sonarr configured in 1 windows 10 pc.
I have a external usb 3 harddrive installed on a 2nd windows 10 pc with plex installed.
all configuration are working fine. 

now i decided to install radarr on the 2nd computer. so it's communicating with sab on the 1st computer.
how do i setup the remote path mapping? is the remote path mapping suppse to point to the default download of sab?

**Comment by jpborjas** [comment]
Exactly, the host is the PC running SAB, Remote Path is the local address for the Sab Movies Folder on the SAB PC, and the Local Path, is the Mapped Movies Folder on the Radarr PC.

**Comment by crockadile35** [comment]
the mapped movie folder on the radarr is where the movies are located and where they are renamed as well?

**Comment by crockadile35** [comment]
if that's the case, it's not importing the movies.
![importing fail](https://cloud.githubusercontent.com/assets/25674133/23096216/87f35688-f5e5-11e6-868d-59e7f2946b11.JPG)

**Comment by crockadile35** [comment]
here is other config pictures of how my setup is, hope someone can help. i uninstall couchpotato completely to concentrate on radarr.
sab client config from radarr
![sab client config](https://cloud.githubusercontent.com/assets/25674133/23096241/39c82bae-f5e6-11e6-8920-6a593e4f042a.JPG)

Local path is where all my movies are. what plex looks at.

**Comment by jpborjas** [comment]
No, you don't specify the folder where your movies are stored, your Plex Movie folder, at least not in the Download Client Screen. You specify that folder when you are adding the movie. 

The Mapped Media Folder is where you can see the Movie Folder on your SAB computer. For instance:

SAB computer is 192.168.11.1 and Radarr computer is .2. On you SAB computer, let's say your Movies folder, where SAB places the movies AFTER they are downloaded, is C:/SAB/Complete, as you said. On your Raddar computer, .2, you have to Map that remote folder so you can see it on your file explorer. I'm a little rusty on Windows (I'm a Mac and Linux user), but you should be able to use the Map option. Let's say you map it as your z: drive, then it'll look like this on .2:

Z:\SAB\Complete

Does that make sense? You should be able to "see" that folder and it's content from your Radarr Computer. If you can, then enter that as your Local Path:

Host: 192.168.11.1
Remote Path: C:/SAB/Complete
Local Path: Z:/SAB/Complete

It is a little confusing and not very intuitive. The good thing is that when it's configured correctly, you'll never touch it again.

**Comment by jpborjas** [comment]
Your Local Path is how your Radarr Computer sees the /SAB/Complete on the remote SAB computer. Nothing to do with the Plex Movies folder. You don't specify the Plex folder in the Download Client Screen at all.

**Comment by hotio** [comment]
@crockadile35 

When SABnzbd has finished downloading files on it's own little PC, it tells Radarr `Hey Radarr! I've got some files and they are stored on my box in this path.`

Somehow these files are accessible to Radarr, except they are not visible on the same path as SABnzbd sees them/reports them.

So now you have to tell Radarr `Hey Radarr! That idiot SABnzbd doesn't know where you are, so his directions are all wrong, so in the future when SABnzbd says turn left, you should really turn right.`

After Radarr has been steered in the right direction, it will find the files and import them into your movie library.

**Comment by jpborjas** [comment]
Hotio, you guys are doing an awesome job. Having said that, I struggled just like @crockadile35 with this configuration. We shouldn't have to specify where SAB stores the file, as it is irrelevant to Radarr. It should be more like "Hey Radarr, remember that movie you asked me to download?, I'm done, and I dropped it in the folder we both agreed on. Go get it using YOUR path.". The directions are not wrong, they are just irrelevant, IMO. 

Awesome product, getting better and better with every release.

Thanks.

**Comment by hotio** [comment]
You are describing more or less "Remote Path Mappings", if they are running on the same system, there's even no need for remote path mappings. I'm not seeing the problem here.

**Comment by crockadile35** [comment]
@jpborjas I did what you said, but I still get import failed, path doesn't exist or is not accessible by radar.

**Comment by crockadile35** [comment]
ok I found the way to do it. since radarr doesn't rename folders yet, it can't find the path which would be the movie plus year. so when sab is downloading a movie like passengers it has a bunch of words after the quality. so I activated the rule in the sorting of sab and I ask sab to rename the folder to "movie name year" now radarr moved the file. so it looks like it works like that. I will do some more test but if it works, then we should tell people to activate the sorting on the sab configuration.

**Comment by jpborjas** [comment]
Well, you do need sorting, to differentiate Movies from TV Shows (different folders), but not renaming. Radarr does the renaming under Settings, Media Management, Movie Naming. Set Rename Movie to Yes, Replace Illegal Characters to Yes, and then use whatever format. I use Movie Title (Release Year). At least you got it to work!

**Comment by crockadile35** [comment]
@jpborjas my tv shows with sonarr is using the drone folder and that is working well. I've tested it both radarr and sonarr the way it is now. the renaming of radarr was set at the beginning it just wasn't working for some reason. 
so now i'm waiting for tonight after some tv show to pass, I will give you an update how I made my setup to work under windows 10. maybe other people having the same issue, can copy my setup if (radarr, sonarr, and sabnzbd) there are no issue between them.

**Comment by crockadile35** [comment]
@jpborjas,
i forgot to mention. Radarr doesn't rename the folders, only the files for now, that's what others said. so i ask sab to rename the folder, and radarr renamed the file with its settings

**Comment by jpborjas** [comment]
@crockadile35 , I was using the Drone Factory Folder as well, and I was having some weird issues and inconsistencies, so I switched to Remote Path Mappings, and the issues went away. If it is working for you, leave it, but I think I'm using the system as intended, and it now makes sense to me. Without getting into a deep explanation here, I came from CouchPotato and SickBeard, and the setting there worked for me, very simply, you drop something in the "monitored folder", and it'll get renamed and imported onto my Plex Folder. Sonarr and Radarr work differently, as it only imports and renames things that it is looking for. I'll leave it at that for the sake of time, but that is how it was explained to me. 

So this is what I'm doing now: Everything that Sonarr and Radarr are looking for, they send a request to SAB, SAB downloads it and let them know it is ready for PostProcess, and the file gets imported and renamed. That is via the Remote Path Mapping. I also use Drone Factory, and that is for the Movies and TV Shows that I download outside of Sonarr and Radarr, but I still have to enter them, so they know what those files are about. 

I hope this helps.
